### PR TITLE
snapps test tool usage doc

### PR DESCRIPTION
**Pull request by deepthiskumar**
New usage doc for snapps test transaction tool

Checklist:

- [ ] Modified the current draft of release notes with details on what is completed or incomplete within this project
- [ ] Document code purpose, how to use it
  - Mention expected invariants, implicit constraints
- [ ] Tests were added for the new behavior
  - Document test purpose, significance of failures
  - Test names should reflect their purpose
- [ ] All tests pass (CI will check this if you didn't)
- [ ] Serialized types are in stable-versioned modules
- [ ] Does this close issues? List them

* Closes #0000

### CODE REVIEW

File: src/app/snapp_test_transaction/README.md [hunk 1]
```diff
@@ -0,0 +1,702 @@
...
+                               (alias: -?)
+```
+
+For example:
+
+```shell
+$mina-snapp-test-transaction create-snapp-account --fee-payer-key my-fee-payer --nonce 0 --receiver-amount 2 --snapp-account-key my-snapp-key
+```
+
+generates the following graphql object- a snapp transaction as input to the sendSnapp mutation. A snapp transaction is basically a list of parties where each [party](https://o1-labs.github.io/snapps-txns-reference-impl/target/doc/snapps_txn_reference_impl/party/index.html) is an update performed on an account.
```
Comment (psteckler) [inline src/app/snapp_test_transaction/README.md:80]: GraphQL

maybe sendSnapp should be in backquotes

File: src/app/snapp_test_transaction/README.md [hunk 2]
```diff
@@ -0,0 +1,702 @@
...
+
+The snapp transaction here has three parties-
+
+1. the fee payer party which specifies who pays the transaction fees and how much
+2. A party that pays the account creation fee to create the new snapp snapp account which in this case is the same as the fee payer
+3. A party to create a new snapp account, set its verification key associated with the test smart contract, and update `editState` and `editSequenceState` permissions to use proofs as [authorization](https://o1-labs.github.io/snapps-txns-reference-impl/target/doc/snapps_txn_reference_impl/party/enum.PartyAuthorization.html).
+
+The authorization used in each of the parties here is a signature of the respective accounts i.e., the updates on these accounts are authorized as per the accounts' permissions.
+
+```json
```
Comment (psteckler) [inline src/app/snapp_test_transaction/README.md:90]: the input isn't JSON, in fact, field names are not double-quoted

I guess just remove the language indicator `json`

File: src/app/snapp_test_transaction/README.md [hunk 3]
```diff
@@ -0,0 +1,702 @@
...
+          permissions:{stake:true,
+            editState:Proof,
+            send:Signature,
+            receive:None,
+            setDelegate:Signature,
+            setPermissions:Signature,
+            setVerificationKey:Signature,
```
Comment (psteckler) [inline src/app/snapp_test_transaction/README.md:203]: I'm thinking about changing all of these to `Proof` for the update-all-fields test, would that be OK?

File: src/app/snapp_test_transaction/README.md [hunk 4]
```diff
@@ -0,0 +1,702 @@
...
+```shell
+$mina-snapp-test-transaction update-state --fee-payer-key my-fee-payer --nonce 2 --snapp-account-key my-snapp-key --fee 5 --snapp-state 1 --snapp-state 2 --snapp-state 3 --snapp-state 4 --snapp-state 5 --snapp-state 6 --snapp-state 7 --snapp-state 8
+```
+
+The snapp transaction here has two parties-
+
+1. The fee payer party which specifies who pays the transaction fees and how much
+2. A party that updates the `app_state` of the snapp account. The authorization required to update the state is a proof (as updated the by deploy-snapp transaction above `editState: Proof`)
+
+```json
```
Comment (psteckler) [inline src/app/snapp_test_transaction/README.md:357]: not JSON

File: src/app/snapp_test_transaction/README.md [hunk 5]
```diff
@@ -0,0 +1,702 @@
...
+}
+  
+```
+
+Account state after the above transaction is included in a block
+
+```json
```
Comment (psteckler) [inline src/app/snapp_test_transaction/README.md:460]: not JSON

File: src/app/snapp_test_transaction/README.md (hunk 3 above)
Comment (psteckler) [reply src/app/snapp_test_transaction/README.md:203]: Or maybe I should have a separate step to update the permissions

**Review by psteckler — APPROVED** [review]
see comments, looks good

File: src/app/snapp_test_transaction/README.md (hunk 3 above)
Comment (deepthiskumar) [reply src/app/snapp_test_transaction/README.md:203]: Would prefer making a separate transaction to update the permissions

**Comment by mrmr1993** [comment]
!ci-build-me

**Pull request merged**